source/vir/src/well_formed.rs

```diff
@@ -306,6 +318,70 @@ fn check_function(ctxt: &Ctxt, function: &Function) -> Result<(), VirErr> {
         }
     }
 
+    if function.x.attrs.integer_ring {
+        if function.x.mode != Mode::Proof {
+            return err_str(&function.span, "integer_ring mode must be declared as proof");
+        }
+        if let Some(body) = &function.x.body {
+            crate::ast_visitor::expr_visitor_check(body, &mut |expr| {
+                match &expr.x {
```

Comment:
we might want to check the function calls in the pre/post conditions don't return bounded integers.

Reply:
Thanks! I added more checks